com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -11,364 +11,217 @@ namespace MLAPI.Prototyping
     [AddComponentMenu("MLAPI/NetworkTransform")]
     public class NetworkTransform : NetworkBehaviour
     {
-        internal class ClientSendInfo
+        public enum Authority
         {
-            public float LastSent;
-            public Vector3? LastMissedPosition;
-            public Quaternion? LastMissedRotation;
+            Server = 0, // default
+            Client,
+            Shared
         }
 
+        [SerializeField, Tooltip("Defines who can update this transform.")]
+        public Authority authority; // todo Luke mentioned an incoming system to manage this at the NetworkBehaviour level, lets sync on this
+
         /// <summary>
         /// The base amount of sends per seconds to use when range is disabled
         /// </summary>
         [Range(0, 120)]
-        public float FixedSendsPerSecond = 20f;
-
-        /// <summary>
-        /// Is the sends per second assumed to be the same across all instances
-        /// </summary>
-        [Tooltip("This assumes that the SendsPerSecond is synced across clients")]
-        public bool AssumeSyncedSends = true;
+        public float FixedSendsPerSecond = 30f;
 
         /// <summary>
+        /// TODO once we have per var interpolation
         /// Enable interpolation
         /// </summary>
+        // ReSharper disable once NotAccessedField.Global
         [Tooltip("This requires AssumeSyncedSends to be true")]
         public bool InterpolatePosition = true;
 
         /// <summary>
+        /// TODO once we have per var interpolation
         /// The distance before snaping to the position
         /// </summary>
+        // ReSharper disable once NotAccessedField.Global
         [Tooltip("The transform will snap if the distance is greater than this distance")]
         public float SnapDistance = 10f;
 
         /// <summary>
+        /// TODO once we have per var interpolation
         /// Should the server interpolate
         /// </summary>
+        // ReSharper disable once NotAccessedField.Global
         public bool InterpolateServer = true;
 
         /// <summary>
+        /// TODO once we have this per var setting
         /// The min meters to move before a send is sent
         /// </summary>
+        // ReSharper disable once NotAccessedField.Global
         public float MinMeters = 0.15f;
```

Comment:
Can we elaborate, maybe with a comment in code on the plan, here? The NetworkTransform would check distance between its value what it last set on the NetworkVar and not update it unless it is large enough ? I'm curious how that plays with snapshot values received from the authoritative server.

Reply:
I was thinking this could be more on the network variable itself. If a server increases a Netvar int by +0.05, the netvar would actually not transmit that info and would wait for the value to be even more different.
The setting in the NetworkTransform would be to just apply it to our netvars when available? What do you think